openpype/hosts/maya/api/plugin.py

```diff
@@ -404,20 +404,59 @@ def create(self, subset_name, instance_data, pre_create_data):
         # A Renderlayer is never explicitly created using the create method.
         # Instead, renderlayers from the scene are collected. Thus "create"
         # would only ever be called to say, 'hey, please refresh collect'
-        self.create_singleton_node()
+        has_singleton = bool(self._get_singleton_node())
+        if not has_singleton:
+            self.create_singleton_node()
+
+        create_mode = pre_create_data.get("create_renderlayer",
+                                          "create_first_renderlayer")
+
+        # Create the renderlayer if user chose layer creation
+        render_layer = None
+        if create_mode in {"create", "create_first_renderlayer"}:
+            rs = renderSetup.instance()
+            existing_layers = rs.getRenderLayers()
+            if create_mode == "create" or not existing_layers:
+                variant = instance_data["variant"]
+                for layer in existing_layers:
+                    if layer.name() == variant:
+                        raise CreatorError(
+                            "Can't create multiple renderlayers of the "
+                            "same variant: {}.\nThis is because the variant "
+                            "defines the renderlayer name.".format(variant)
+                        )
 
-        # if no render layers are present, create default one with
-        # asterisk selector
-        rs = renderSetup.instance()
-        if not rs.getRenderLayers():
-            render_layer = rs.createRenderLayer("Main")
-            collection = render_layer.createCollection("defaultCollection")
-            collection.getSelector().setPattern('*')
+                render_layer = rs.createRenderLayer(variant)
+                collection = render_layer.createCollection("defaultCollection")
+                collection.getSelector().setPattern('*')
+
```

Comment:
If I'm looking correctly, the last `else:` can be moved here, and all the `if has_singleton` could be removed.
```suggestion
        if not has_singleton:
            # Collect all, including any renderlayers that might have existed.
            # By doing this we make sure that whenever anyone creates a render
            # instance and already has renderlayers present in their scene
            # these renderlayers will directly show up in the list on create.
            # (This is crucial because the singleton node itself does not show)
            # By RenderLayerCreator.create we make it so that the renderlayer
            # instances directly appear even though it just collects scene
            # renderlayers. This doesn't actually 'create' any scene contents.
            # Since renderlayers might have changed (manually by user) we want
            # to make sure we list all of them that haven't been listed yet.
            self.collect_instances()
            return
```

Reply:
I actually think the current logic is correct.

If there's currently a singleton (meaning it has likely already been collected) then we only want to register the new layer if any was created.

If no singleton existed before, we now want to collect ALL renderlayers since that's what would be collected on a reset as well. As such, the logic I believe is correct currently.

Reply:
@iLLiCiTiT bumping this, so we can test it after this is resolved.